packages/core/js-client/src/jsServiceHost/serviceUtils.ts

```diff
@@ -65,7 +65,23 @@ export const getParticleContext = (
 export function registerDefaultServices(peer: FluencePeer) {
   Object.entries(builtInServices).forEach(([serviceId, service]) => {
     Object.entries(service).forEach(([fnName, fn]) => {
-      peer.internals.regHandler.common(serviceId, fnName, fn);
+      const wrapped = async (req: CallServiceData) => {
+        const res = await fn(req);
+
+        if (
+          res.retCode === ResultCodes.error &&
+          typeof res.result === "string"
```

Comment:
And what should happen if `res.retCode === ResultCodes.error` but `typeof res.result !== "string"` ? Is this at all possible. If not - why not?